Source/WebCore/platform/gstreamer/GStreamerQuirkBroadcomBase.cpp

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright (C) 2024 Igalia S.L
+ * Copyright (C) 2024 Metrological Group B.V.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public License
+ * aint with this library; see the file COPYING.LIB.  If not, write to
+ * the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301, USA.
+ */
+
+#include "config.h"
+#include "GStreamerQuirkBroadcom.h"
+
+#include <string.h>
+
+#if USE(GSTREAMER)
+
+#include "GStreamerCommon.h"
+
+namespace WebCore {
+
+GST_DEBUG_CATEGORY_STATIC(webkit_broadcom_base_quirks_debug);
+#define GST_CAT_DEFAULT webkit_broadcom_base_quirks_debug
+
+GStreamerQuirkBroadcomBase::GStreamerQuirkBroadcomBase()
+{
+    GST_DEBUG_CATEGORY_INIT(webkit_broadcom_base_quirks_debug, "webkitquirksbroadcombase", 0, "WebKit Broadcom Base Quirks");
+}
+
+bool GStreamerQuirkBroadcomBase::queryBufferingPercentage(MediaPlayerPrivateGStreamer* playerPrivate, const char*& elementName, GRefPtr<GstQuery>& query) const
```

Comment:
Maybe, but it would make the logic in https://github.com/WebKit/WebKit/pull/29998/commits/0aae941a74fbc7be84ce9ddffa73b7779bff0388#diff-4376b854280205494dc9de0f08f985afd4d6584a7487c0a136adac2954c003cdR1340 more complex. That elementName is a variable that comes from the caller site and has its own interpretation and usage rules, as well as query.

Reply:
Can't you do a `value_or` at the caller?

Reply:
Ended up using `const char*` and nullptr to signal "query not successful".